spongeauth/api/views.py

```diff
@@ -73,6 +74,11 @@ def _create_user(request):
     verified = request.POST.get('verified', 'false') == 'true'
     dummy = request.POST.get('dummy', 'false') == 'true'
 
+    try:
+        accounts.models.validate_username(username)
```

Comment:
I'd rather you run all model validation here: basically, before `.save` is called, you should call `.full_clean()`, which will do all the validation (including username validation):

https://docs.djangoproject.com/en/2.1/ref/models/instances/#django.db.models.Model.full_clean

Your test shouldn't need to change.